src/hotkey/command_executor.cpp

```diff
@@ -77,9 +77,47 @@ static void event_execute(const SDL_Event& event, command_executor* executor);
 
 bool command_executor::execute_command(const hotkey_command&  cmd, int /*index*/, HOTKEY_EVENT_TYPE type)
 {
-	if (type == HOTKEY_EVENT_RELEASE)
-		return false; // nothing responds to a release yet
+	if (type == HOTKEY_EVENT_RELEASE) {
+		switch(cmd.id) {
+			// release a scroll key, un-apply scrolling in the given direction
+			case HOTKEY_SCROLL_UP:
+				keyboard_scroll(0, 1);
+				break;
+			case HOTKEY_SCROLL_DOWN:
+				keyboard_scroll(0, -1);
+				break;
+			case HOTKEY_SCROLL_LEFT:
+				keyboard_scroll(1, 0);
+				break;
+			case HOTKEY_SCROLL_RIGHT:
+				keyboard_scroll(-1, 0);
+				break;
+			default:
+				return false; // nothing else handles a hotkey release
+		}
+
+		return true;
+	}
+
+	// special handling for scroll keys, which do not handle repeat
```

Comment:
yeah, this part was one of the messiest. I guess I could just drop the special casing for repeat and handle it in `keyboard_scroll` since every other hotkey handles repeat.